#!/usr/bin/env python

PACKAGE = 'gazelle_sim'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

rbt_enum = gen.enum( [ 
        gen.const('diff_circ',int_t, 0, 'Differenial Steer / Circluar Base'),
        gen.const('diff_rect',int_t, 1, 'Differenial Steer / Rectangular Base'),
        gen.const('ackermann',int_t, 2, 'Ackermann Steer / Rectangular Base')
    ], 'Robot type')




gen.add('robot_name',    str_t, 0, 'Robot Name', 'robot0')
gen.add('robot_type',    int_t, 0, 'Robot Type', 0, 0, 2, edit_method=rbt_enum)

gen.add('length',       double_t, 0, 'Body length (Rect)', 0.0,   0.0,  5.0)
gen.add('width',        double_t, 0, 'Body width (Rect)',  0.0,   0.0,  5.0)
gen.add('radius',       double_t, 0, 'Body radius (Circ)', 0.2,   0.1,  2.0)
gen.add('wheelbase',    double_t, 0, 'Wheelbase (Ack)',    0.2,   0.1,  2.0)
gen.add('max_steer',    double_t, 0, 'Max Steer Ang (Ack)',0.5,   0.4,  0.8)
gen.add('trackwidth',   double_t, 0, 'Trackwidth',         0.2,   0.1,  2.0)
gen.add('axle_dist',    double_t, 0, 'Distance to axle',   0.0,  -2.0,  2.0)
gen.add('rwheel',       double_t, 0, 'Wheel radius',       0.0,  0.05,  0.5)
gen.add('wheel_width',  double_t, 0, 'Wheel width',        0.0,  0.02,  0.25)

gen.add('camera_x',     double_t, 0, 'Camera x location',  -2.0,   0.0,  2.0)
gen.add('camera_y',     double_t, 0, 'Camera y location',  -2.0,   0.0,  2.0)
gen.add('camera_z',     double_t, 0, 'Camera z location',   1.0,   0.0,  2.0)
gen.add('camera_phi',   double_t, 0, 'Camera angle',      0.785,   0.0,  1.57)
gen.add('camera_f',     double_t, 0, 'Camera focal length', 1,    700.0, 10000.0)
gen.add('camera_width', double_t, 0, 'Camera icon width',  0.05,   0.0,  0.2)
gen.add('camera_length',double_t, 0, 'Camera icon length',  0.1,   0.0,  0.2)

gen.add('lidar_x',      double_t, 0, 'Lidar x location',   -2.0,   0.0,  2.0)
gen.add('lidar_y',      double_t, 0, 'Lidar y location',   -2.0,   0.0,  2.0)
gen.add('lidar_r',      double_t, 0, 'Lidar icon radius',   0.0,   0.0,  2.0)

gen.add('x_init',     double_t, 0, 'Initial x position', 0.0, -10.0, 10.0)
gen.add('y_init',     double_t, 0, 'Initial y position', 0.0, -10.0, 10.0)
gen.add('theta_init', double_t, 0, 'Initial angle'     , 0.0, -6.28, 6.28)


exit(gen.generate(PACKAGE, 'gazelle_sim', 'GazelleRobot'))
